### PR TITLE
Copter - continuously evaluate prearm checks

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -12,24 +12,18 @@ void AP_Arming_Copter::update(void)
...
 
 // performs pre-arm checks and arming checks
 bool AP_Arming_Copter::all_checks_passing(bool arming_from_gcs)
 {
-    if (pre_arm_checks(true)) {
-        set_pre_arm_check(true);
-    } else {
-        return false;
-    }
+    set_pre_arm_check(pre_arm_checks(true));
```
Comment (magicrub) [inline ArduCopter/AP_Arming.cpp]: Is this backwards?

Comment (peterbarker) [reply ArduCopter/AP_Arming.cpp]: Sadly, no, I don't think so.

set_pre_arm_checks sets some variables based on the result of pre_arm_checks.

The argument to pre_arm_checks is whether to be noisy or not.

**Comment by OXINARF** [comment]
@peterbarker Like, like, like! 😄 Has this been discussed with @rmackay9? Is the WIP because it hasn't been tested yet?

**Comment by peterbarker** [comment]
@OXINARF Sorry, it was WIP because I created the PR while on the road away from a PX4, so couldn't test it on HW.  I've since done so and forgotten to remove the WIP flag :-)

@rmackay9 has previous indicated support for getting rid of the latching.

There are more patches to come along here (in a different PR!), but they will involve behaviour changes.  For example, sometimes we honour the arming check bits on the safety switch, sometimes we don't.

**Comment by amilcarlucas** [comment]
ping

**Comment by amilcarlucas** [comment]
It works but has some issues with range finder noise, it keeps playing tunes constantly although the copter is just sitting still on the ground.

**Comment by peterbarker** [comment]
@amilcarlucas Well, that's certainly interesting!

Can it be tuned out with the rangefinder settings?

The rangefinder should be consistently healthy when on the ground, shouldn't it?

**Comment by amilcarlucas** [comment]
@peterbarker : I'll test disabling the rangefinder to see if it helps, but for now I just reverted your first commit and it works like a charm :)

**Comment by amilcarlucas** [comment]
I will test the rangefinder issue soonish. But it should be fixed in the range finder code. This PR is fine.

**Comment by rmackay9** [comment]
Merged, thanks!
One small additional change we could make is to completely remove the copter.ap.pre_arm_check flag.

**Pull request merged**